### PR TITLE
Bugfix/hide toggle elem on load master

### PR DESCRIPTION
**Pull request by joefitter**
This fix is necessary as the addition of panel currently only occurs on click, also the aria-controls tag was pointing at the wrong element

### CODE REVIEW

**Comment by joefitter** [comment]
@gavboulton @easternbloc @JoeChapman @HughePaul @oli-weighell2

File: test/client/spec/spec.progressive-reveal.js [hunk 1]
```diff
@@ -39,11 +39,15 @@ describe('Progressive Reveal', function () {
...
             beforeEach(function () {
-                $('form').append('<div id="check-toggle-panel" class="reveal js-hidden">');
+                $('form').append('<div id="check-toggle-panel">');
                 $('label').append('<input type="checkbox" id="check" name="check" data-toggle="check-toggle">CheckBox');
                 progressiveReveal();
             });
 
+            it('should have added the js-hidden class', function () {
+                $('#check-toggle-panel').hasClass('js-hidden').should.be.ok;
+            });
```
Comment (gavboulton) [inline test/client/spec/spec.progressive-reveal.js]: Do we need this test in other parts too?

File: test/client/spec/spec.progressive-reveal.js [hunk 2]
```diff
@@ -39,11 +39,15 @@ describe('Progressive Reveal', function () {
         describe('parent panel', function () {
 
             beforeEach(function () {
-                $('form').append('<div id="check-toggle-panel" class="reveal js-hidden">');
+                $('form').append('<div id="check-toggle-panel">');
                 $('label').append('<input type="checkbox" id="check" name="check" data-toggle="check-toggle">CheckBox');
                 progressiveReveal();
             });
 
+            it('should have added the js-hidden class', function () {
```
Comment (gavboulton) [inline test/client/spec/spec.progressive-reveal.js]: What about 'adds the `js-hidden` class'?

File: assets/javascript/progressive-reveal.js [hunk 3]
```diff
@@ -35,8 +32,14 @@ function inputClicked(e, target) {
...
-    var toggleId = input.getAttribute(toggleAttr),
-        toggle = document.getElementById(toggleId);
+    var element = input.getAttribute(toggleAttr);
+    // check if the element supplied is part of an {id}-panel
+    // if so then toggle this parent element to also toggle
+    // associated labels and legends
+    var toggleId = document.getElementById(element + '-panel')
+        ? element + '-panel'
+        : element;
+    var toggle = document.getElementById(toggleId);
```
Comment (gavboulton) [inline assets/javascript/progressive-reveal.js]: Good spot on this. Can you add a test.

**Comment by joefitter** [comment]
@gavboulton updated

File: test/client/spec/spec.progressive-reveal.js [hunk 4]
```diff
@@ -151,6 +163,11 @@ describe('Progressive Reveal', function () {
                     progressiveReveal();
                 });
 
+                it('hides #radio1-toggle and #radio2-toggle', function () {
```
Comment (gavboulton) [inline test/client/spec/spec.progressive-reveal.js]: This doesn't make sense when viewed in the console.

File: test/client/spec/spec.progressive-reveal.js [hunk 5]
```diff
@@ -174,10 +191,14 @@ describe('Progressive Reveal', function () {
                     // group 2
                     $('form').append('<label for="radio4">');
                     $('label[for=radio4]').append('<input type="radio" name="group2" id="radio4" data-toggle="radio4-toggle">');
-                    $('form').append('<div id="radio4-toggle" class="reveal js-hidden">');
+                    $('form').append('<div id="radio4-toggle">');
                     progressiveReveal();
                 });
 
+                it('hides #radio4-toggle', function () {
```
Comment (gavboulton) [inline test/client/spec/spec.progressive-reveal.js]: This doesn't make sense when viewed in the console.

**Comment by joefitter** [comment]
@gavboulton updated

File: .gitignore [hunk 6]
```diff
@@ -1,5 +1,6 @@
 assets/stylesheets/*.css
 assets/images/icons
+assets/images/govuk
```
Comment (gavboulton) [inline .gitignore]: Is this a hangover from something else?

File: test/client/spec/spec.progressive-reveal.js [hunk 7]
```diff
@@ -39,20 +43,24 @@ describe('Progressive Reveal', function () {
...
+                $('form').append('<div id="check-toggle-panel">');
                 $('label').append('<input type="checkbox" id="check" name="check" data-toggle="check-toggle">CheckBox');
                 progressiveReveal();
             });
 
+            it('adds the js-hidden class', function () {
+                $('#check-toggle-panel').hasClass('js-hidden').should.be.ok;
+            });
+
+            it('doesn\'t hide #show-toggle', function () {
```
Comment (gavboulton) [inline test/client/spec/spec.progressive-reveal.js]: Using div IDs in descriptions doesn't describe the functionality.

File: assets/javascript/progressive-reveal.js [hunk 8]
```diff
@@ -12,11 +12,8 @@ function inputClicked(e, target) {
     target = target || helpers.target(e);
     var shown;
     _.each(groups[target.name], function (input) {
-        var id = input.getAttribute(toggleAttr);
-        // check if the element supplied is part of an {id}-panel
-        // if so then toggle this parent element to also toggle
-        // associated labels and legends
-        var toggle = document.getElementById(id + '-panel') || document.getElementById(id);
+        var id = input.getAttribute('aria-controls');
```
Comment (gavboulton) [inline assets/javascript/progressive-reveal.js]: There isn't a test checking that the `aria-controls` attribute is added. If we're relying on this elsewhere we should add a test.

File: .gitignore (hunk 6 above)
Comment (joefitter) [inline .gitignore]: if you freshly clone the repo and npm install, these are added as untracked files so need to be gitignored

**Comment by joefitter** [comment]
@gavboulton I'll update but please can you leave all these comments at once so we can reduce the lead time to fixes?

**Comment by gavboulton** [comment]
hmpo-frontend-toolkit@6.0.1 published.

**Pull request merged**